### DISCUSSION TITLE
[SYCL][DPCPP] WSL/WSL2 and Intel HD Graphics

### DISCUSSION THREAD

**Discussion by vrpascuzzi** [microsoft/WSL#6338] [Q&A]
**System**
  * [Razer Blade Studio 2020](https://www.razer.com/gaming-laptops/Razer-Blade-15/RZ09-0330QEM3-R3U1?gclid=Cj0KCQiA8dH-BRD_ARIsAC24umbOeLOkTF6HH2S9nCNj2TyrOx_ej7jHpwN1041pCNXQKo9e-_EMR4kaAgQFEALw_wcB)
  * Edition:	Windows 10 Pro Insider Preview
Version:	2004
Installed on:	‎12/‎11/‎2020
OS build:	21277.1000
Experience:	Windows Feature Experience Pack 220.32303.0.2003
WSL kernel:	5.4.72-microsoft-standard-WSL2

**Problem**
Intel HD Graphics setup/configuration in WSL/WSL2

**Process**
[Installed NVIDIA CUDA Quadro for WSL drivers](https://developer.nvidia.com/cuda/wsl/download) for Windows 10 and followed [instructions for CUDA on WSL](https://docs.nvidia.com/cuda/wsl-user-guide/index.html#abstract) to configure an Ubuntu distro. Built and ran a simple CUDA app -- everything works fine. Instructions from NVIDIA are very clear.
Intel is a different story. [Installed Intel beta drivers](https://downloadcenter.intel.com/download/29526) and followed [Intel's GPGPU setup instructions for Ubuntu 20.04](https://dgpu-docs.intel.com/installation-guides/ubuntu/ubuntu-focal.html). Built a simple app with [oneAPI](https://www.oneapi.com/) went smoothly but ran into a runtime error when selecting Intel iGPU:
```
$ ./test_sycl
terminate called after throwing an instance of 'cl::sycl::runtime_error'
  what():  No device of requested type available. Please check https://software.intel.com/en-us/articles/intel-oneapi-dpcpp-compiler-system-requirements-beta -1 (CL_DEVICE_NOT_FOUND)
Aborted
```
Selecting `default`/`CPU` device things are fine:
```
$ ./test_sycl
Running on: Intel(R) Core(TM) i7-10875H CPU @ 2.30GHz
<...>
```
I get the same result when using WSL and WSL2.

Anyone able to get Intel HD Graphics (iGPU) working in WSL/WSL2?

**Comment by alvarovm** [comment]
Is this still an issue in Oct/2022?

**Reply by vrpascuzzi** [reply]
@alvarovm,

Thanks for the reply. Does WSL now support Intel HD Graphics?

**Reply by alvarovm** [reply]
@vrpascuzzi  It seems that at least with Level Zero runtime it is possible. I made some notes for myself here: https://gist.github.com/alvarovm/9f15e7db2390e840d5508d67fd4fc99e , check them out.
